src/Spago/Packages.hs

```diff
@@ -85,27 +85,48 @@ getProjectDeps :: Spago m => Config -> m [(PackageName, Package)]
 getProjectDeps Config{..} = getTransitiveDeps packageSet dependencies
 
 
+fst' :: (a, b, c) -> a
+fst' (x, _, _) = x
+
+snd' :: (a, b, c) -> b
+snd' (_, x, _) = x
+
+trd' :: (a, b, c) -> c
+trd' (_, _, x) = x
+
 -- | Return the transitive dependencies of a list of packages
 --   Code basically from here:
 --   https://github.com/purescript/psc-package/blob/648da70ae9b7ed48216ed03f930c1a6e8e902c0e/app/Main.hs#L227
 getTransitiveDeps :: Spago m => PackageSet -> [PackageName] -> m [(PackageName, Package)]
 getTransitiveDeps PackageSet{..} deps = do
   echoDebug "Getting transitive deps"
-  Map.toList . fold <$> traverse (go Set.empty) deps
+  let xs  = map (go Set.empty Set.empty Set.empty) deps
+
+  let cycleErrors = foldMap trd' xs
+  let notFoundErrors = foldMap snd' xs
+  let packageMap = Map.toList $ foldMap fst' xs
+
```

Comment:
It'd be cool if we could avoid doing the fold-map-dance three times like above 🤔